### PR TITLE
Add some pinned deps to runtime envs

### PR DESCRIPTION
**Pull request by nateraw**
Here I'm adding a few helpful deps for videos...

- `decord` a fast video decoder. probably very helpful for folks to make submissions faster.
- `pytorchvideo` - facebookresearch library containing a bunch of helpful video utilities
- `av` helpful reader/writer (used by pytorchvideo). Is a little slower than decord but has more features/I've found it to be a bit more stable in my experience

...as well as adding:
- `timm`, which has a bunch of pretrained vision models folks may want to leverage
- `transformers`, which has vision models as well (notably some nice video models I was playing with)

I also pinned the version of `pytorch-lightning` because lightning flash had a loose version of it in its requirements.

I didn't build this on CPU or GPU...I read there were GH actions so I'm hoping I can just work against that 😉 

Happy to discuss if you folks don't want some of these in the environment!

### CODE REVIEW

**Comment by chrisjkuch** [comment]
Hey @nateraw - thanks for the PR! It looks like the build is failing because of a conflict:
```
The conflict is caused by:
    The user requested pytorchvideo==0.1.5
    lightning-flash[image,video] 0.8.1 depends on pytorchvideo==0.1.2; extra == "video"
```
Have you tested the build locally?

P.S. We are still undergoing internal discussions about your forum question, hoping to have more for you sooner rather than later.

**Comment by chrisjkuch** [comment]
@nateraw Another failure:
```
The conflict is caused by:
    The user requested timm==0.6.12
    lightning-flash[image,video] 0.8.1 depends on timm>=0.4.5; extra == "image"
    segmentation-models-pytorch 0.3.1 depends on timm==0.4.12
    The user requested timm==0.6.12
    lightning-flash[image,video] 0.8.1 depends on timm>=0.4.5; extra == "image"
    segmentation-models-pytorch 0.3.0 depends on timm==0.4.12
    The user requested timm==0.6.12
    lightning-flash[image,video] 0.8.1 depends on timm>=0.4.5; extra == "image"
    segmentation-models-pytorch 0.2.1 depends on timm==0.4.12
    The user requested timm==0.6.12
    lightning-flash[image,video] 0.8.1 depends on timm>=0.4.5; extra == "image"
    segmentation-models-pytorch 0.2.0 depends on timm==0.4.12
```
Would you mind debugging this locally? It will probably be much, much faster to just create a fresh conda environment and attempt to install it locally, since you benefit from package caching that doesn't happen here.

**Comment by nateraw** [comment]
ah darn. will take a look. Flash has a bit of a messy nest of deps, so will have to see if we can work around it. Folks are using that dep actively?

**Comment by chrisjkuch** [comment]
TBH I'm not sure - I don't think any of the submission so far have included code execution components.

**Comment by nateraw** [comment]
Can we remove flash in this PR then for now and cross the bridge if needed later? 🤞 

Just ran some tests...all in new conda envs. flash installs 57 deps with those extras. 25 deps with the vanilla install. Compared to others (transforms: 11, pytorchvideo: 11, timm: 10, ...) that's a LOT imo for something we dunno if we need

**Comment by chrisjkuch** [comment]
FWIW I pinned `timm==0.4.12` and built a new env locally and it seemed to work - does that version suffice for your needs?

**Comment by nateraw** [comment]
I think there are some newer models (+ accompanying pretrained weights) in `timm` that folks would want to maybe leverage, which is why I opted for newer version.

but it doesn't effect me right now so if ya want to do that, that's cool too

**Comment by nateraw** [comment]
but you will run into this problem again! haha

**Comment by chrisjkuch** [comment]
> Can we remove flash in this PR then for now and cross the bridge if needed later?

Let me give this some thought / discussion and get back to you

**Comment by nateraw** [comment]
All good! Also FWIW the lightning team moves quick so maybe someone from their side can remedy this for us (pinging @borda)

**Comment by chrisjkuch** [comment]
Another avenue - looks like the offending flash dependency [segmentation_models.pytorch](https://github.com/qubvel/segmentation_models.pytorch) just recently upgraded `timm` to 0.6.12 but has yet to cut a new release

**Comment by nateraw** [comment]
maybe @qubvel can make a release? 👀 would be much appreciated :)

**Comment by chrisjkuch** [comment]
> maybe qubvel can make a release? 👀 would be much appreciated :)

Looks like we'll get one shortly, let's try building again once it comes in.

**Comment by Borda** [comment]
fyi, some compatibility is rolling over here: https://github.com/Lightning-AI/lightning-flash/pull/1487

**Comment by chrisjkuch** [comment]
Alright @nateraw, pinning  `segmentation-models-pytorch==0.3.2` seems to solve all the outstanding issues - can you make the change & I'll run the builds and we can merge?

File: runtime/environment-cpu.yml [hunk 1]
```diff
@@ -34,5 +35,10 @@ dependencies:
   - tslearn=0.5.2
   - pip: 
     - augly[image,video]==1.0.0
+    - decord==0.6.0
     - lightning-flash[video,image]==0.8.1
-    - tensorflow-cpu==2.9.1
\ No newline at end of file
+    - pytorch-lightning==1.8.6
+    - pytorchvideo==0.1.2
```
Comment (chrisjkuch) [inline runtime/environment-cpu.yml:41]:
```suggestion
    - pytorchvideo==0.1.2
    - segmentation-models-pytorch==0.3.2
```

File: runtime/environment-gpu.yml [hunk 2]
```diff
@@ -36,4 +37,8 @@ dependencies:
   - pip: 
     - augly[image,video]==1.0.0
     - lightning-flash[video,image]==0.8.1
-    - tensorflow-gpu==2.9.1
\ No newline at end of file
+    - pytorch-lightning==1.8.6
+    - pytorchvideo==0.1.2
```
Comment (chrisjkuch) [inline runtime/environment-gpu.yml:41]:
```suggestion
    - pytorchvideo==0.1.2
    - segmentation-models-pytorch==0.3.2
```

**Review by chrisjkuch — CHANGES_REQUESTED** [review]

**Comment by nateraw** [comment]
@chrisjkuch you know if we could use the latest pytorchvideo in this case then? Or would we still be constrained to 1.2.0?

Sorry for late reply btw - in France for hugging face offsite so low availability.

**Comment by chrisjkuch** [comment]
Yes, I think we're still constrained to `pytorchvideo==1.2.0` because of the flash dependency. If 1.5.0 is absolutely necessary, we can consider removing flash, but if we can help it I'd rather not.

**Comment by nateraw** [comment]
Got it. having a look now!

**Review by chrisjkuch — APPROVED** [review]
alright we're building and good to go

**Comment by nateraw** [comment]
Thanks!!! Trained a model this past weekend - excited to submit :) haha hopefully > 0.0 score this time.

**Pull request merged**